build/nginx/Dockerfile

```diff
@@ -0,0 +1,88 @@
+FROM node:14-bullseye-slim as base
+
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
+    git \
+    build-essential \
+    python \
+    ca-certificates \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /root/frappe-bench
+RUN mkdir -p sites/assets
+
+ARG FRAPPE_VERSION
+RUN git clone --depth 1 -b ${FRAPPE_VERSION} https://github.com/frappe/frappe apps/frappe
+
+
+
+FROM base as frappe_prod_node_modules
+
+# Install production node modules
+RUN yarn --cwd apps/frappe --prod
+
+
+
+FROM frappe_prod_node_modules as frappe_assets
+
+# Install development node modules
+RUN cd apps/frappe && \
+    if [ "$(uname -m)" = "aarch64" ]; then \
+    yarn remove svg-sprite || true \
+    && yarn add sass; \
+    fi \
+    && yarn
+
+# Build assets they're stored in frappe-bench/sites/assets
+RUN echo "frappe" >sites/apps.txt \
+    && yarn --cwd apps/frappe run production \
+    && rm sites/apps.txt
+
+
+
+FROM base as erpnext_prod_node_modules
+
+ARG ERPNEXT_VERSION
+RUN git clone --depth 1 -b ${ERPNEXT_VERSION} https://github.com/frappe/erpnext apps/erpnext
+
+RUN yarn --cwd apps/erpnext --prod
+
+
+
+FROM erpnext_prod_node_modules as erpnext_assets
+
+RUN yarn --cwd apps/erpnext
+
+COPY --from=frappe_assets /root/frappe-bench/apps/frappe/node_modules /root/frappe-bench/apps/frappe/node_modules
+COPY --from=frappe_assets /root/frappe-bench/apps/frappe/package.json /root/frappe-bench/apps/frappe/yarn.lock /root/frappe-bench/apps/frappe/
+
+RUN echo "frappe\nerpnext" >sites/apps.txt \
+    && yarn --cwd apps/frappe run production --app erpnext \
+    && rm sites/apps.txt
+
+
+
+FROM base as error_pages
+
+RUN git clone --depth 1 https://github.com/frappe/bench /root/bench
+
+
+
+FROM nginx:1.21-alpine as frappe
+
+COPY --from=error_pages /root/bench/bench/config/templates/502.html /usr/share/nginx/html
+COPY --from=base /root/frappe-bench/apps/frappe/frappe/public /usr/share/nginx/html/assets/frappe
+COPY --from=frappe_prod_node_modules /root/frappe-bench/apps/frappe/node_modules /usr/share/nginx/html/assets/frappe/node_modules
+COPY --from=frappe_assets /root/frappe-bench/sites /usr/share/nginx/html
+
+COPY nginx-template.conf /
```

Comment:
Yep. See https://github.com/vrslev/frappe_docker/blob/5e84d7734c362d86e95b05dc3da72b548082f3c2/build/nginx/Dockerfile#L80
Also, it it okay that container uses `nginx` image instead of `nginxinc/nginx-unprivileged`?

Reply:
not really,

if you try to make normal nginx image run as non root then it won't work. it needs additional configuration.

not good to run apps in non root containers.

Reply:
> not good to run apps in non root containers.

It is good to run app in non-root containers — less privileges.

So, I should use `nginxinc/nginx-unprivileged`, right?

Reply:
Yes like before, with following

https://github.com/frappe/frappe_docker/blob/c2b15431281e470678676bd6b1d9ac0288536850/build/frappe-nginx/Dockerfile#L52-L52

Reply:
I updated compose file, nginx template and Dockerfile. I used `1101` user instead of `nginx` and it works without changes use suggested:
https://github.com/vrslev/frappe_docker/blob/225c69709780e7dd2e282aafd9e16e63654d3ed1/build/nginx/Dockerfile#L71-L90

Reply:
https://github.com/frappe/frappe_docker/pull/307 this needed the users to match.
It was for issue https://github.com/frappe/frappe_docker/pull/299

I don't think it exists anymore. Can you confirm it didn't introduce back 299

Reply:
I will later check if cache works without changes.

Reply:
It did not work without changes! It works now: https://github.com/frappe/frappe_docker/pull/617/commits/075df53a337e234ad297b8e84ef583be3ab7f25e. Can you review this?